src/tools/rbd_mirror/instance_watcher/Types.h

```diff
@@ -39,8 +39,110 @@ WRITE_CLASS_ENCODER(PeerImageId);
 
 typedef std::set<PeerImageId> PeerImageIds;
 
+enum NotifyOp {
+  NOTIFY_OP_IMAGE_ACQUIRE  = 0,
+  NOTIFY_OP_IMAGE_RELEASE  = 1,
+};
+
+struct ImageAcquirePayload {
+  static const NotifyOp NOTIFY_OP = NOTIFY_OP_IMAGE_ACQUIRE;
+
+  uint64_t request_id;
+  std::string global_image_id;
+  PeerImageIds peers;
```

Comment:
Agreed -- definitely won't hurt. I'm just trying to say that passing around a collection of peers on each image acquire/release probably doesn't matter. If we pass only the single peer that was associated with the original event, perhaps that could provide some optimization in the future so perhaps it doesn't hurt to replace the collection w/ a single peer. If we don't end up using it, at least it can be useful for debugging events.